### DISCUSSION TITLE
Can this package be used in replacement of redis ?

### DISCUSSION THREAD

**Discussion by ceessay** [typegoose/mongodb-memory-server#466] [Q&A] [answered]
## Versions

- NodeJS: 0.0.0
- mongodb-memory-server-*: 0.0.0
- mongodb: 0.0.0
- mongoose: 0.0.0 
- system: 

package: mongo-memory-server 

## What is your question?

I have to process large set of arrays coming from MongoDb atlas. I use json files to store temporary results of my processing. I would like to know if I could use mongo-memory-server instead of redis to store my temporary  results

**Comment by hasezoey** [comment]
this package here (`mongodb-memory-server-core`) only does download & manage an `mongod` (mongodb) instance, it will definitely be slower than redis (redis is built with in-memory in mind, mongodb not)

its like manually starting an local instance of `mongod` along side your project, you need to decide yourself if mongodb is right for this job

**Comment by jhunterkohler** [comment, marked as answer]
It's a terrible idea. Redis is an entirely different database paradigm. MongoDB documents will always be slower than a key/pair database lookup table, especially one as optimized as redis. Understand that they are fundamentally different in database design, and not just in the fact that redis runs direct from memory. 

If you wan't to learn about this general idea look at this video, I saw it earlier today:
https://www.youtube.com/watch?v=W2Z7fbCLSTw&t=4s
